From e0a7a3a3a253a7360556cfbfc2054e7e3795f38f Mon Sep 17 00:00:00 2001
From: Alexey Puzhevich <alexey.puzhevich@intel.com>
Date: Mon, 16 Jun 2025 15:32:35 +0300
Subject: [PATCH] [JSW-934] jgs: preboot: run in simulation environment

The needed changes were introduced to enable ppboot
execution in simulation environment (Coral).

Change-Id: I3fb5441b3c379216b7dee2b658e3e8ca6d828830
Signed-off-by: Alexey Puzhevich <alexey.puzhevich@intel.com>
---

diff --git a/arch/jgs/arc/common.c b/arch/jgs/arc/common.c
index 683f4ed..96a5401 100644
--- a/arch/jgs/arc/common.c
+++ b/arch/jgs/arc/common.c
@@ -271,7 +271,7 @@
 	hl_dev_pll_init();
 }
 
-#define SPSRAM_ADDR_MASK	0xFFFFF
+#define SPSRAM_ADDR_MASK	0xFFFF
 
 static u32 jgs_translate_arc_addr(u64 addr)
 {
diff --git a/arch/jgs/arc/common.h b/arch/jgs/arc/common.h
index df35551..5a35c02 100644
--- a/arch/jgs/arc/common.h
+++ b/arch/jgs/arc/common.h
@@ -12,16 +12,19 @@
 
 #define HL_FW_BOOT_PTRS_BASE		0
 
-/* same as IO0_SP_SRAM_FULL_BASE (left here for backwards compatibility) */
-#define HL_SP_SRAM_BASE			0x300007FF8A00000ULL
+/* same as IO0_SP_SRAM_FULL_BASE */
+/* TODO: redesign include tree to use definition from specs */
+#define HL_SP_SRAM_BASE			0x300007FF9DF0000ULL
 #define HL_SP_SRAM_SIZE			SZ_1M
 
 /* CFG is at Arc region F, specs CFG base address is 0x70000000. 7+8 = 0xF */
 /* defined here in addition to specs because they are in use im common code */
 #define HL_CFG_BASE			0x80000000
-#define FW_MEM_REGION_BASE		0x78800000 /* IO Die0 Half0 FW memories */
-#define FW_SRAM_BASE			(FW_MEM_REGION_BASE + 0x1A0000)
-#define SP_SRAM_BASE			(FW_MEM_REGION_BASE + 0x200000)
+/* TODO: redesign include tree to use definition from specs */
+#define FW_MEM_REGION_BASE		0x78640000 /* IO Die0 Half0 FW memories */
+#define FW_SRAM_BASE			(FW_MEM_REGION_BASE)
+/* TODO: redesign include tree to use definition from specs */
+#define SP_SRAM_BASE			0x79DF0000
 
 #define BOOT_INFO_BASE			SP_SRAM_BASE
 #define BOOT_INFO_SIZE			SZ_32K
